lib/optimize/CommonsChunkPlugin.js

```diff
@@ -2,183 +2,189 @@
 	MIT License http://www.opensource.org/licenses/mit-license.php
 	Author Tobias Koppers @sokra
 */
-var nextIdent = 0;
+"use strict";
+let nextIdent = 0;
 
-function CommonsChunkPlugin(options) {
-	if(arguments.length > 1) {
-		throw new Error("Deprecation notice: CommonsChunkPlugin now only takes a single argument. Either an options " +
-			"object *or* the name of the chunk.\n" +
-			"Example: if your old code looked like this:\n" +
-			"    new webpack.optimize.CommonsChunkPlugin('vendor', 'vendor.bundle.js')\n\n" +
-			"You would change it to:\n" +
-			"    new webpack.optimize.CommonsChunkPlugin({ name: 'vendor', filename: 'vendor.bundle.js' })\n\n" +
-			"The available options are:\n" +
-			"    name: string\n" +
-			"    names: string[]\n" +
-			"    filename: string\n" +
-			"    minChunks: number\n" +
-			"    chunks: string[]\n" +
-			"    children: boolean\n" +
-			"    async: boolean\n" +
-			"    minSize: number\n");
-	}
-	if(Array.isArray(options) || typeof options === "string") {
-		options = {
-			name: options
-		};
-	}
-	this.chunkNames = options.name || options.names;
-	this.filenameTemplate = options.filename;
-	this.minChunks = options.minChunks;
-	this.selectedChunks = options.chunks;
-	if(options.children) this.selectedChunks = false;
-	this.async = options.async;
-	this.minSize = options.minSize;
-	this.ident = __filename + (nextIdent++);
-}
+class CommonsChunkPlugin {
+	constructor(options) {
+		if(arguments.length > 1) {
+			throw new Error(`Deprecation notice: CommonsChunkPlugin now only takes a single argument. Either an options
```

Comment:
@Kovensky I thinks it's better to leave it as is (i.e. string concatenation) than this for better readability